File: backend/src/main/java/com/nanmuli/blog/domain/tag/Tag.java
package com.nanmuli.blog.domain.tag;

import com.nanmuli.blog.shared.domain.BaseAggregateRoot;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@EqualsAndHashCode(callSuper = false)
public class Tag extends BaseAggregateRoot<Long> {
    private static final long serialVersionUID = 1L;

    private String name;
    private String slug;
    private String color;
    private String icon;
    private String description;
    private Integer articleCount;
    private Integer status;
}

File: frontend/src/types/tag.ts
// 注意：ID 使用 string 类型，因为 JavaScript number 无法精确表示 64 位整数
export interface Tag {
  id: string
  name: string
  slug: string
  color: string
  icon: string
  description?: string
  articleCount: number
  status: number
  createTime: string
  updateTime: string
}
